core/src/main/scala/io/finch/internal/ToService.scala

```diff
@@ -29,9 +29,18 @@ private[finch] final class ToService[A, CT <: String](
     to
   }
 
-  def apply(req: Request): Future[Response] = underlying(Input.request(req)) match {
-    case EndpointResult.Matched(rem, out) if rem.isEmpty =>
-      out.map(oa => copyVersion(req, oa.toResponse(tr, tre))).run
-    case _ => Future.value(Response(req.version, Status.NotFound))
+  private[this] def setCurrentTime(rep: Response): Response = {
```

Comment:
You can move this machinery inside the `copyVersion` method (and rename it accordingly to something like `conformHttp`) so we can avoid that extra `map` at the end.